rp2040-hal/src/async.rs

```diff
@@ -0,0 +1,18 @@
+macro_rules! block_on {
+    ($me:ident, $is_ready:ident ($($arg:expr),*), $setup_flags:ident ($($arg2:expr),*)) => {
+        future::poll_fn(|cx| {
```

Comment:
Uh, I didn't know about `poll_fn`. TIL!
That makes https://github.com/jannic/nb-async-adapter/blob/main/src/lib.rs mostly obsolete, I guess :smile: